examples/cluster/plot_hdbscan.py

```diff
@@ -0,0 +1,211 @@
+utility functions for convenience.
+"""
+# %%
+import numpy as np
+
+from sklearn.cluster import HDBSCAN, DBSCAN
+from sklearn.datasets import make_blobs
+import matplotlib.pyplot as plt
+
+
+def plot(X, labels, probabilities=None, parameters=None, ground_truth=False, ax=None):
+    if ax is None:
+        _, ax = plt.subplots(figsize=(10, 4))
+    labels = labels if labels is not None else np.ones(X.shape[0])
+    probabilities = probabilities if probabilities is not None else np.ones(X.shape[0])
+    # Black removed and is used for noise instead.
+    unique_labels = set(labels)
+    colors = [plt.cm.Spectral(each) for each in np.linspace(0, 1, len(unique_labels))]
+    # The probability of a point belonging to its labeled cluster determines
+    # the size of its marker
+    proba_map = {idx: probabilities[idx] for idx in range(len(labels))}
+    for k, col in zip(unique_labels, colors):
+        if k == -1:
+            # Black used for noise.
+            col = [0, 0, 0, 1]
+
+        class_index = np.where(labels == k)[0]
+        for ci in class_index:
+            ax.plot(
+                X[ci, 0],
+                X[ci, 1],
+                "x" if k == -1 else "o",
+                markerfacecolor=tuple(col),
+                markeredgecolor="k",
+                markersize=4 if k == -1 else 1 + 5 * proba_map[ci],
+            )
+    n_clusters_ = len(set(labels)) - (1 if -1 in labels else 0)
+    preamble = "True" if ground_truth else "Estimated"
+    title = f"{preamble} number of clusters: {n_clusters_}"
+    if parameters is not None:
+        parameters_str = ", ".join(f"{k}={v}" for k, v in parameters.items())
+        title += f" | {parameters_str}"
+    ax.set_title(title)
+    plt.tight_layout()
+
+
+# %%
+# Generate sample data
+# --------------------
+# One of the greatest advantages of HDBSCAN over DBSCAN is its out-of-the-box
+# robustness. It's especially remarkable on heterogenous mixtures of data.
+# Like DBSCAN, it can model arbitrary shapes and distributions, however unlike
+# DBSCAN it does not require specification of an arbitray (and indeed tricky)
+# `eps` hyperparameter. For example, below we generate a dataset composed of
+# a mixture of three diagonal Gaussians.
+centers = [[1, 1], [-1, -1], [1.5, -1.5]]
+X, labels_true = make_blobs(
+    n_samples=750, centers=centers, cluster_std=[0.4, 0.1, 0.75], random_state=0
+)
+plot(X, labels=labels_true, ground_truth=True)
+# %%
+# Scale Invariance
+# -----------------
+# It's worth remembering that, while DBSCAN provides a default value for `eps`
+# parameter, it is entirely meaningless and must be tuned for your specific
+# dataset. As a simple demonstration, consider what happens when we find an
+# epsilon value that works for one dataset, and try to apply it to a
+# similar but rescaled versions of the dataset. Below are plots of the original
+# dataset, and versions rescaled by 0.5 and 3 respectively.
```

Comment:
Nit: let's be gentle with $\varepsilon$, it did nothing wrong.

```suggestion
# It's worth remembering that, while DBSCAN provides a default value for `eps`
# parameter, it hardly has a proper default value and must be tuned for the
# specific dataset at use.
# As a simple demonstration, consider the clustering for a `eps` value tuned
# for one dataset, and clustering obtained with the same value but applied to
# rescaled versions of the dataset. Below are plots of the original
# dataset, and versions rescaled by 0.5 and 3 respectively.
```

Also, I think it would be nice to have the rescaling information in the title of plots. This would allow removing the last sentence of this paragraph.